registered.go

```diff
@@ -0,0 +1,30 @@
+package gokore
+
+import (
+	"fmt"
+	"sort"
+)
+
+var registered Endpoints = nil
+
+func RegisterEndpoint(endpoint EndpointInterface) {
+	registered = append(registered, endpoint)
+	sort.Sort(EndpointsByTitle(registered))
+}
+
+func GetEndpointCount() int { return len(registered) }
+func GetEndpointByIndex(index int) (EndpointInterface, error) {
+	if 0 <= index && index < len(registered) {
+		return registered[index], nil
+	} else {
+		return nil, fmt.Errorf("Index is out of bounds.")
+	}
+}
+func GetEndpointByTitle(title string) (EndpointInterface, error) {
+	for _, endpoint := range registered {
```

Comment:
I'm pretty sure most of the "internal implementation" will be rewritten many times before we ship. Public interface might even change a few times as well. So, I don't think it's really important right now how it is implemented.